packages/suite/src/components/suite/Preloader/Preloader.tsx

```diff
@@ -69,6 +71,21 @@ export const Preloader = ({ children }: PreloaderProps) => {
         return <InitialLoading timeout={90} />;
     }
 
+    if (
+        router.route?.app !== 'settings' &&
+        selectedDevice?.features &&
+        selectedDevice.connected === true &&
+        !isFirmwareRevisionCheckDisabled
+    ) {
+        const failedChecks = Object.values(selectedDevice.authenticityChecks).filter(
+            check => check.success === false,
+        );
+
+        if (failedChecks.length > 0) {
+            return <DeviceCompromised />;
```

Comment:
- [ ] Todo: send to Sentry/Analytics that this happened (check with Matěj, consult with Device Auth Check code how it does it)